# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/vlang/no_builtin_update.patch
# Copyright (C) 2025 - 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- v-0.5.2.orig/cmd/v/v.v	2026-07-13 00:47:19.000000000 +0200
+++ v-0.5.2/cmd/v/v.v	2026-09-26 16:34:22.562519724 +0200
@@ -42,13 +42,11 @@ const external_tools = [
 	'repl',
 	'repeat',
 	'retry',
-	'self',
 	'setup-freetype',
 	'shader',
 	'share',
 	'should-compile-all',
 	'sqlite',
-	'symlink',
 	'scan',
 	'test',
 	'test-all', // runs most of the tests and other checking tools, that will be run by the CI
@@ -59,7 +57,6 @@ const external_tools = [
 	'time',
 	'timeout',
 	'tracev',
-	'up',
 	'vet',
 	'wipe-cache',
 	'watch',
--- v-0.5.2.orig/vlib/v/help/default.txt	2026-07-13 00:47:19.000000000 +0200
+++ v-0.5.2/vlib/v/help/default.txt	2026-09-26 16:34:16.217648755 +0200
@@ -63,10 +63,6 @@ V supports the following commands:
                                the documentation, testing, etc.
 
 * Installation Management Utilities:
-  symlink                      Create a symbolic link for V.
-  up                           Run the V self-updater.
-  self [-prod]                 Run the V self-compiler, use -prod to optimize
-                               compilation.
   version                      Print the version text and exits.
 
 * Package Management Utilities:
--- v-0.5.2.orig/vlib/v/pref/pref.v	2026-07-13 00:47:19.000000000 +0200
+++ v-0.5.2/vlib/v/pref/pref.v	2026-09-26 16:34:16.217767902 +0200
@@ -516,6 +516,16 @@ pub fn parse_args_and_show_errors(known_
 	}
 
 	mut no_skip_unused := false
+	
+	$if tools_vflags ? {
+		flag := $d('tools_vflags', '')
+		for def in flag.split(' ') {
+			res.parse_define(def)
+		}
+	} $else {
+		$compile_warn("no tools_vflags set")
+	}
+
 	mut command, mut command_idx := '', 0
 	mut build_vsh_source := false
 	for i := 0; i < args.len; i++ {
